File: data/snippets/github.com/khilnani/snippets/vs/compile.php
#!/usr/bin/php
<?php

function compile($expr) {
  echo "Preparing ...\n";

  system("rm -rf classes");
  system("mkdir classes");

  $cmd = "";

  if($expr == "*") {
    $cmd = "scalac -d classes src/*.scala";
  } else if( strpos( $expr, ".scala") > -1) {
    $cmd = "scalac -d classes $expr";
  } else {
    $cmd = "scalac -d classes $expr.scala";
  }

  $cmd .= " -feature";

  echo "Compiling $cmd\n";
  system( $cmd );

  echo "Done.\n";
}

if( count($argv) == 1) {
  echo "No file specified. Using *\n";
  compile("*");
} else
  compile ($argv[1]);
?>

File: data/snippets/github.com/khilnani/snippets/vs/compile.py
#!/usr/bin/python

import sys
import os

def compile (expr="*"):
  print "Preparing ..." 

  os.system( "rm -rf classes" )
  os.system( "mkdir classes" )

  cmd = ''

  if expr == "*":
    cmd = "scalac -d classes src/*.scala"
  elif expr.endswith( ".scala" ):
    cmd = "scalac -d classes " + expr
  else:
    cmd = "scalac -d classes " + expr + ".scala" 

  cmd += " -feature"

  print "Compiling: " + cmd
  os.system( cmd )

  print "Done."

if len(sys.argv) < 2:
  print "No File specified. Using *"
  compile()
else:
  compile (sys.argv[1])
